SwiftUIBasic/SwiftUIBasic/Authentication/ViewModel/AuthenticationViewModel.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  AuthenticationViewModel.swift
+//  SwiftUIBasic
+//
+//  Created by Jaesung Lee on 2021/06/14.
+//
+
+import Combine
+import Commons
+import SendBirdCalls
+
+class AuthenticationViewModel: ObservableObject {
+    @Published var userID: String = ""
+    @Published var accessToken: String = ""
+    @Published var error: String = ""
+    @Published var authRequired: Bool = true
+    
+    var currentUser: User? { SendBirdCall.currentUser }
+    
+    func authenticate() {
+        let authParams = AuthenticateParams(
+            userId: userID,
+            accessToken: accessToken.isEmpty ? nil : accessToken
+        )
+        
+        SendBirdCall.authenticate(with: authParams) { [self] user, error in
+            if let error = error {
+                self.error = error.localizedDescription
```

Comment:
Dont need `self` here

Reply:
I used self because of the same naming (self.error and error declared in closure)

Reply:
Removed